Added App Introduction Screens

legendgroupv2 opened the pull request:
Added three screens explaining what Meshtastic is, how to use the app and how to set up encryption. 
This is only shown on the first launch of the app - after completing the introduction the app returns to normal functionality.

Closes #333 

<details><summary>Screenshots</summary>

![Screenshot_20220810-171356_Meshtastic](https://user-images.githubusercontent.com/73121875/183961057-92895141-45a0-421a-b452-5a85e70f61c2.jpg)
![Screenshot_20220810-171400_Meshtastic](https://user-images.githubusercontent.com/73121875/183961090-9304b352-ac6d-42c9-ae60-997a9035f920.jpg)
![Screenshot_20220810-171404_Meshtastic](https://user-images.githubusercontent.com/73121875/183961112-b28e1e72-c9c2-4993-91f1-2290c9436391.jpg)

</details>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/meshtastic/Meshtastic-Android?pullRequest=460) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/meshtastic/Meshtastic-Android?pullRequest=460) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/meshtastic/Meshtastic-Android?pullRequest=460) it.</sub>

andrekir commented on app/src/main/res/values/strings.xml:
```diff
@@ -144,4 +144,13 @@
+    <string name="intro_get_started">...Let\'s get started!</string>
+    <string name="intro_started_text">Connect your meshtastic device by using either bluetooth or serial. \n\nYou may need to wake your node if they are battery powered. \n\nYou can see which devices are compatible at www.meshtastic.org/docs/hardware</string>
+    <string name="intro_ecryption_title">"Setting up encryption"</string>
+    <string name="intro_encryption_text">To enable encryption, go to the channel tab and change the channel name. This will set a random key and enable AES256 encryption. \n\nTo communicate with other devices they will need to scan your QR code or follow the shared link to configure the channel settings.</string>
```
the default channel has a public AES256 key, but encryption is enabled. maybe "to create your own channel and encryption key" ...

legendgroupv2 replied:
Have reworded it to reflect your suggestions. Thanks :)

andrekir commented on app/src/main/res/values/strings.xml:
```diff
@@ -144,4 +144,13 @@
+    <string name="intro_welcome_title">Welcome to Meshtastic</string>
+    <string name="intro_meshtastic_desc">Meshtastic is an open-source, off-grid, encrypted communication platform. The meshtastic radios form a mesh network and communicate using the LoRa protocol to send text messages</string>
+    <string name="intro_get_started">...Let\'s get started!</string>
+    <string name="intro_started_text">Connect your meshtastic device by using either bluetooth or serial. \n\nYou may need to wake your node if they are battery powered. \n\nYou can see which devices are compatible at www.meshtastic.org/docs/hardware</string>
```
- app now supports BLE, Serial and HTTP.
- "You may need to wake your node if they are battery powered." no longer applies.

```suggestion
    <string name="intro_started_text">Connect your meshtastic device by using either Bluetooth, Serial or WiFi. \n\nYou can see which devices are compatible at www.meshtastic.org/docs/hardware</string>
```

andrekir reviewed (changes requested):
@legendgroupv2 Thank you for your contribution!
Please check my reviews 👍

legendgroupv2 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -1,196 +1,186 @@
-        </receiver>
-    </application>
-
-</manifest>
```
Apologies it seems like my IDE reformatted the file so my changes aren't clear. I've reverted the file back to the original, committed it, and then added my code to the latest commit. It's the activity relating to AppIntroduction. Hopefully, this resolves the issue.

legendgroupv2 commented:
Thank you for your reviews. I've addressed them all and committed the changes. In regards to the AndroidManifest file, it seems like my IDE reformatted the file so my changes aren't very clear. I've reverted the file back to the original, committed it, and then added my code to the latest commit. Hopefully, this resolves the issue.

andrekir commented:
I took care of some leftovers in the manifest. if you run into this again, a tip is to revert at the file level on your local copy. you can review the diffs in the IDE after. 👍

andrekir reviewed (approved):
all good! thank you!
